classes/migration/upgrade/PKPv3_3_0UpgradeMigration.inc.php

```diff
@@ -334,24 +335,53 @@ private function _migrateSubmissionFiles() {
 		// Create entry in files and revisions tables for every submission_file
 		import('lib.pkp.classes.file.FileManager');
 		$fileManager = new FileManager();
-		$rows = Capsule::table('submission_files')
-			->join('submissions', 'submission_files.submission_id', '=', 'submissions.submission_id')
-			->orderBy('file_id')
-			->orderBy('revision')
+		$rows = Capsule::table('submission_files', 'sf')
+			->join('submissions AS s', 'sf.submission_id', '=', 's.submission_id')
+			->leftJoin('review_round_files AS rrf', function (JoinClause $j) {
+				$j->on('rrf.submission_id', '=', 'sf.submission_id')
+					->on('rrf.file_id', '=', 'sf.file_id')
+					->on('rrf.revision', '=', 'sf.revision');
+			})
+			->orderBy('sf.file_id')
+			->orderBy('rrf.review_round_id')
+			->orderBy('sf.revision')
 			->get([
-				'context_id',
-				'file_id',
-				'revision',
-				'submission_files.submission_id',
-				'genre_id',
-				'file_type',
-				'file_stage',
-				'date_uploaded',
-				'original_file_name'
+				's.context_id',
+				'sf.file_id',
+				'sf.revision',
+				'sf.submission_id',
+				'sf.genre_id',
+				'sf.file_type',
+				'sf.file_stage',
+				'sf.date_uploaded',
+				'sf.original_file_name',
+				'rrf.review_round_id'
 			]);
 		$fileService = Services::get('file');
 		$submissionFileService = Services::get('submissionFile');
+		$lastFileId = null;
+		$lastReviewRoundId = null;
+		$lastInsertedFileId = Capsule::table('submission_files')->max('file_id');
+		$fileIdMap = [];
 		foreach ($rows as $row) {
+			// Due to the new database structure, a file_id that spans across N+1 review rounds needs to be forked to a new file_id, to avoid being consolidated later on the upgrade process, which would cause data loss
+			// Statistics, logs, etc. will keep being referenced by the "main" file_id, which is not ideal, but also not possible to address.
+			// This branch will be accessed once the review_round_id gets changed within a group of the same file_ids
+			if ($lastFileId === $row->file_id && $lastReviewRoundId !== $row->review_round_id) {
```

Comment:
The first combination of `file_id` + `review_round_id` is supposed to be ok! I mean, it's not needed to be updated.
We just have an issue when the same `file_id` is shared across a different `review_round_id`, then we need to decouple it.